packages/dev/core/src/Compute/computeShader.ts

```diff
@@ -402,8 +416,12 @@ export class ComputeShader {
             }
         }
 
-        this._engine.computeDispatch(this._effect, this._context, this._bindings, x, y, z, this._options.bindingsMapping, this.gpuTimeInFrame);
-
+        if ("buffer" in options) {
+            const buffer = options.buffer instanceof StorageBuffer ? options.buffer.getBuffer() : options.buffer;

```

Comment:
You can use `ComputeShader._BufferIsDataBuffer` to avoid using `instanceof` (see my code above).